src/utils/sources-tree/updateTree.js

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+
+import { Set } from "immutable";
+
+import type { Props, State } from "./types";
+
+import { addToTree } from "./addToTree";
+import { collapseTree } from "./collapseTree";
+import { createParentMap } from "./utils";
+
+function compareProps(nextProps: Props, props: Props) {
+  const next = Set(nextProps.sources.valueSeq());
+  const prev = Set(props.sources.valueSeq());
+  return next.subtract(prev);
```

Comment:
ooh - nice find w/ Immutable.Set.subtract

I think we can simplify this and save a bunch of CPU cycles :clock:

`return next.sources.subtract(prev.sources)`

Reply:
`next` and `prev` are the sourceMaps already, so I don't think it goes any deeper.